db/views/searches_v02.sql

```diff
@@ -0,0 +1,15 @@
+SELECT
+  app.id,
+  app_ver.id as application_version_id,
+  app_ver.application ->> 'laa_reference' as laa_reference,
+  app_ver.application -> 'firm_office' ->> 'name' as firm_name,
+  (app_ver.application -> 'defendant' ->> 'first_name') || ' ' || (app_ver.application -> 'defendant' ->> 'last_name') as client,
+  app_ver.search_fields,
+  app.has_been_assigned_to,
+  app.created_at as date_submitted,
+  app.updated_at as date_updated, -- events can happen in caseworker after this? should we be checking max event time as well???
```

Comment:
Events aren't a separate table, so every time an event is added to the event array `updated_at` will be bumped, but it won't be the timestamp of the last update, it will be the timestamp of the last time an event completed syncing via a background job. So ideally yeah, you'd use max event time OR app_ver.created_at, whichever is later

Reply:
Will look into this a bit more. Means more gnarly SQL i am guessing.

Reply:
I have analysed the data on production and found that of 219 submissions:

#### number of submissions with greater than 60 sec diff between event and updated_at
- 17 over a minute difference between app updated and latest event time
- [only] 3 if excluding those updated on 1 July 2024 by a backfill operation
- app_updated_at is always later than latest event if there is a diff.

#### number of submissions with greater than 60 sec diff between event and app_ver.created
- 23 over a minute difference
- 20 excluding those updated on 1 July 2024 by a backfill operation
- [only] 4 where app_ver_created is actually present  i.e. need to backfill app_ver timestamps